spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/AutoConfigurations.java

```diff
@@ -51,7 +50,7 @@ protected Collection<Class<?>> sort(Collection<Class<?>> classes) {
 		List<String> sorted = SORTER.getInPriorityOrder(names);
 		return sorted.stream()
 			.map((className) -> ClassUtils.resolveClassName(className, null))
-			.collect(Collectors.toCollection(ArrayList::new));
+			.collect(Collectors.toUnmodifiableList());
```

Comment:
If we're going to change this, it should use `Stream`'s `toList()` rather than `collect`. That said, it's technically public API so there's a (very slight) chance of a breaking change and I'm not sure it's worth changing at all.

Reply:
`Stream.toList()` can't compile here.

Reply:
PR updated, I reverted this change.

Reply:
Gah, generics are a pain. Please revert this change.

Reply:
Already reverted.